conans/client/manager.py

```diff
@@ -327,21 +327,27 @@ def _load_deps_info(self, current_path, conanfile, output):
             except:
                 output.error("Parse error in '%s' file in %s" % (BUILD_INFO, current_path))
         else:
-            output.warn("%s file not found in %s\nIt is recommended for source, build and package "
-                        "commands\nYou can generate it using 'conan install -g env -g txt'"
-                        % (BUILD_INFO, current_path))
-
-        env_file = os.path.join(current_path, CONANENV)
-        if os.path.exists(env_file):
-            try:
-                deps_env_info = DepsEnvInfo.loads(load(env_file))
-                conanfile.deps_env_info = deps_env_info
-            except:
-                output.error("Parse error in '%s' file in %s" % (CONANENV, current_path))
-        else:
-            output.warn("%s file not found in %s\nIt is recommended for source, build and package "
-                        "commands\nYou can generate it using 'conan install -g env -g txt'"
-                        % (CONANENV, current_path))
+            if not for_imports:
+                output.warn("%s file not found in %s\nIt is recommended for source, build and package "
+                            "commands\nYou can generate it using 'conan install -g env -g txt'"
+                            % (BUILD_INFO, current_path))
+            else:
+                raise ConanException("%s file not found in %s\nIt is required for imports "
```

Comment:
@annulen I have changed the output.error => ConanException, because it is totally impossible that the imports will work without this information, so the error is better captured and conan invocation returns error, instead of just printing an error message.